torchchat/generate.py

```diff
@@ -353,27 +353,34 @@ def prefill(
         width = x.size(1)
         assert input_pos.size(0) == width
 
-        if batch is not None:
+        if self.model.config.model_type == ModelType.Flamingo:
+            assert batch is not None, "Flamingo requires batch"
+
             # TODO: Verify sequential prefill works with multimodal models
-            tokens = batch["tokens"]
+            is_multimodal = True
             if 'encoder_input' in batch:
                 encoder_input = batch['encoder_input']
+                encoder_mask = batch["encoder_mask"]
+                is_multimodal = True
             else:
                 encoder_input = None
+                encoder_mask = None
+                is_multimodal = False
 
-            seq_len = tokens.size(1)
+            seq_len = x.size(1)
```

Comment:
What's the difference between tokens and x?

Reply:
No difference, all equal to text input
Previously we pull the text input out of `batch` inside prefill and call it `tokens`
Now we pull out text input first then forward it into prefill function and call it `x`
The reason we change the name is other single modality models usually use `x` to represent the text token inputs.